lib/flight_scheduler/event_processor.rb

```diff
@@ -177,6 +187,13 @@ def cancel_job(job)
   end
   module_function :cancel_job
 
+  def job_timed_out(job_id)
+    job = FlightScheduler.app.job_registry.lookup(job_id)
+    return unless job
+    job.state = job.terminal_state? ? 'TIMEOUT' : 'TIMINGOUT'
```

Comment:
Why is a job being changed from a terminal state?

Reply:
Because daemon side the order `JOB_TIMED_OUT` and `NODE_DEALLOCATED` in is undefined. It is conceivable possible that `JOB_TIMED_OUT` is received after the de-allocation.

In this case it would have already transitioned into a terminal state, but is actually `TIMEOUT`. Whilst this isn;t ideal, I couldn't think of a way to prevent it without additional a confirmation message.